### ISSUE TITLE
String choices cannot be described

### ISSUE THREAD

**Issue by rowanc1** [3ptscience/properties#103]
We do not currently have the ability to store descriptions on the StringChoice property.

```
"choices": [
    "C",
    "N",
    "F"
],
"descriptions": [
    "Cell centred data",
    "Nodal data",
    "Face data"
]
```

**Comment by bsmithyman** [comment]
Cool. I think only one description for choices that accept multiple validation inputs?

**Comment by fwkoch** [comment]
Would these descriptions be used exclusively for documentation? eg:
```
Required
    Properties
        location: (StringChoice) choices: 'C', 'N', 'F' ('cell-centred data', 'nodal data', 'face data', respectively)
```
As described above, this would make sets and dictionaries invalid inputs since they are unordered. There could be checks in place to ensure choices is ordered if descriptions are provided. Alternatively, descriptions could be a dictionary with choice/description pairs...

We could also keep the dictionary implementation as-is and just modify the docstring creation (and maybe write some best-practices documentation on StringChoice). I'm thinking something like:
```
locations = properties.StringChoice(
    doc='data location',
    choices={
        'cell-centred data': ['C', 'CC'],
        'nodal data': 'N',
        'face data': 'F'
    }
)
```
with doc something like:
```
Required
    Properties
        location: (StringChoice) choices:  cell-centred data ('C', 'CC'), nodal data ('N'), face data ('F')
```
Then someone could `'cell-centred data'`, `'C'`, and `'CC'` would all be valid inputs, but intended usage would just be 'C' and 'CC'. Perhaps we could modify validation so the values are coerced to the first value in the choices dictionary rather than the key (ie 'C' rather than 'cell-centred data').

**Comment by rowanc1** [comment]
I think that the order is occasionally important.

Can we add `descriptions` to the `StringChoice` independently? Perhaps as a dictionary?

```python
locations = properties.StringChoice(
    doc='data location',
    choices=['C', 'N'],
    descriptions={'C': 'cell-centred data', 'N': 'nodal data'}
)
```

**Comment by fwkoch** [comment]
Yeah, that would be fine. Allows for both ordered and unordered choices.

**Comment by fwkoch** [comment]
Implemented in #120
